test/dotnet.Tests/CommandTests/ToolManifestTests.cs

```diff
@@ -364,6 +367,153 @@ public void DifferentVersionOfManifestFileItThrows()
+            _fileSystem.File.WriteAllText(manifestFile, _jsonContent);
+
+            var toolManifest = new ToolManifestFinder(new DirectoryPath(_testDirectoryRoot), _fileSystem);
+
+            PackageId packageId = new PackageId("dotnetsay");
+            NuGetVersion nuGetVersion = NuGetVersion.Parse("2.1.4");
+            Action a = () => toolManifest.Add(new FilePath(manifestFile),
+                packageId,
+                nuGetVersion,
+                new[] { new ToolCommandName("dotnetsay") });
+
+            a.ShouldNotThrow();
+
+            _fileSystem.File.ReadAllText(manifestFile).Should().Be(_jsonContent);
+        }
+
+        [Fact]
+        public void GivenAnInvalidManifestFileOnSameDirectoryWhenAddItThrows()
+        {
+            string manifestFile = Path.Combine(_testDirectoryRoot, _manifestFilename);
+            _fileSystem.File.WriteAllText(manifestFile, _jsonWithInvalidField);
+
+            var toolManifest = new ToolManifestFinder(new DirectoryPath(_testDirectoryRoot), _fileSystem);
+
+            PackageId packageId = new PackageId("dotnetsay");
+            NuGetVersion nuGetVersion = NuGetVersion.Parse("3.0.0");
+            Action a = () => toolManifest.Add(new FilePath(manifestFile),
+                packageId,
+                nuGetVersion,
+                new[] {new ToolCommandName("dotnetsay")});
+
+            a.ShouldThrow<ToolManifestException>()
+                .And.Message.Should().Contain(
+                    string.Format(LocalizableStrings.InvalidManifestFilePrefix,
+                        manifestFile,
+                        string.Empty));
+
+            _fileSystem.File.ReadAllText(manifestFile).Should().Be(_jsonWithInvalidField);
+        }
+
+        [Fact]
+        public void GivenManifestFileOnSameDirectoryItCanFindTheFirstManifestFile()
+        {
+            string manifestPath = Path.Combine(_testDirectoryRoot, _manifestFilename);
+            _fileSystem.File.WriteAllText(manifestPath, _jsonContent);
+            var toolManifest = new ToolManifestFinder(new DirectoryPath(_testDirectoryRoot), _fileSystem);
+            FilePath toolmanifestFilePath = toolManifest.FindFirst();
+
+            toolmanifestFilePath.Value.Should().Be(manifestPath);
+        }
+
+        [Fact]
+        public void GivenManifestFileOnSameDirectoryItDoesNotThrowsWhenTheManifestFileIsNotValid()
+        {
+            string manifestPath = Path.Combine(_testDirectoryRoot, _manifestFilename);
+            _fileSystem.File.WriteAllText(manifestPath, _jsonWithMissingField);
+            var toolManifest = new ToolManifestFinder(new DirectoryPath(_testDirectoryRoot), _fileSystem);
+            FilePath toolmanifestFilePath = toolManifest.FindFirst();
+
+            toolmanifestFilePath.Value.Should().Be(manifestPath);
+        }
+
+        [Fact]
+        public void GivenManifestFileOnSameDirectoryItThrowsWhenTheManifestFileCannotBeFound()
```

Comment:
Are we adding tests for when the manifest is not on the same directory?

Reply:
The test name is outdated, let me remove the "same directory". In the beginning i did want "add" to do "find the file" and then "and". Later i realize i need to surface the path of the manifest file location anyway. I made it 2 steps, "add" and "find" are 2 APIs. Add will throw if it is not pointing to a valid file

Reply:
I think it might be hard to do automatically 

`Add(1,2,3,4,5....81);` will be formatted to the following by my editor current rule
```
Add(1,2,3,4,5....80,
           81); 
```

but if we enforce one line per param, it will look like
```
Add(
  1,
  2,
  3,
  .
  .
  .
  81
)
```

Reply:
fixed